src/libs/ActiveClientManager/index.js

```diff
@@ -7,6 +7,16 @@ const clientID = Str.guid();
 const maxClients = 20;
 
 let activeClients;
+let setReady = null;
+
+// Whether the client Manager has started
+let isInit = false;
+
+// Are we ready to determine active Client?
+const isReady = new Promise((res) => {
```

Comment:
These changes are needed as the Client manager incorrectly reports the leadership of the client

Reply:
Can you maybe elaborate on this?

Reply:
Yeah sure. There are cases where we run important callbacks once but It is possible that the client is not set before that callback call. Thus a promise to make sure that we run the callback. We have to use a flag and combination of Promises to achieve this.
as 
```
function init() {
    Onyx.merge(ONYXKEYS.ACTIVE_CLIENTS, [clientID]);
    isInit = true;
}
```
The above method does not really initialize the client it just request Onyx to set the client id. `merge` Ops is async and I think it does not return thenable callback thus we have to go to `connect` call.

```
 if (isInit) {
            setReady();
        }
```
Now we mark is ready by resolving the promise as we have set the clientId on the client stack.        This connect could fire multiple times, we only resolve the promise when `isInit` is set.

Reply:
Would it be possible to wait until the client _is_ set to do the rest of the logic here?

One idea would be to add an `onReady` function to `ActiveClientsManager` so that we can register a callback then wait to do the `Onyx.connect()` below?

I think it might make it more obvious that we are waiting for the active client to be set

```js
ActiveClientsManager.onReady(() => {
    Onyx.connect({
        key: ONYXKEYS.NETWORK_REQUEST_QUEUE,
        callback: ...
    })
});
```

Reply:
Ok Thanks I am seeing the problem now. And yeah, Onyx being asynchronous is making this difficult once again 😓